lib/ci-helper.ts

```diff
@@ -763,6 +772,211 @@ export class CIHelper {
+                                    date
+                                    email
+                                    name
+                                    user {
+                                        login
+                                    }
+                                }
+                            }
+                        }
+                    },
+                    url
+                    files(last:20) {
+                        totalCount,
+                        nodes {
+                            additions,
+                            deletions,
+                            path
+                        }
+                    }
+                }
+            }
+        }`;
+
+        type queryResponse = {
+            repository: Repository;
+        }
+
+        const repository =
+            await this.github.graphql<queryResponse>(
+            pr.baseOwner,
+            query
+        );
+
+        if (!repository || !repository.repository.pullRequest) {
+            throw new Error(`Pull request not found.`);
+        }
+        const prInfo = repository.repository.pullRequest;
+
+        if (prInfo.closed) {
+            return false;           // nothing to see here
+        }
+
+        const genLabel = (label: string, message: string): labelInfo => {
+            return {label, message};
+        };
+
+        try {
+            const commits = prInfo.commits.nodes;
+
+            if (commits) {
+                commits.map(node => {
+                    const login = node?.commit.committer?.user?.login ??
+                        node?.commit.author?.user?.login;
+
+                    // login may not be present if email unknown to github
+                    if (login && pr.author !== login) {
+                        throw new Error("Request author is not commit author.");
+                    }
+                });
+            }
+
+            const files = prInfo.files;
+            if (files && files.totalCount && files.nodes) {
+                files.nodes.map(node => {
+                    const changedFile = node as PullRequestChangedFile;
+                    this.checkNewUserFile(changedFile, files.totalCount,
+                        genLabel);
+                });
+            } else {
+                throw new Error("No files updated.");
+            }
+
+            const queryRef = `query {
+                repository(name: "${pr.baseRepo}", owner: "${pr.author}") {
+                    ref(qualifiedName: "${prInfo.baseRefName}") {
+                        name
+                    }
+                }
+            }`;
+
+            const refInfo = await this.github.graphql<queryResponse>(
+                pr.baseOwner,
+                queryRef
+            );
+
+            if (!refInfo || !refInfo.repository.ref) {
+                throw new Error(`Base branch ${prInfo.baseRefName
+                    } not found in ${pr.author}/${pr.baseRepo}.`);
+            }
```

Comment:
I guess this check should come before any linting, as a deleted fork is pretty fatal: that's not a contribution, that's a "throw over the wall".

Reply:
It is last to avoid a second trip to the server.  It was not happening in many of the tests against existing closed bogus PRs because they failed for other reasons.  If they don't have the branch there is a good chance they will fail the author check.

Reply:
Since this affects new users only, I think it'd not be too bad to do this check first. After all, if the user already deleted their fork, the PR can be closed without any further effort spent: In this case, the contributor is clearly not interested in collaborating, so why should the Git project be.